### PR TITLE
Calculator logic milestone 4 operate and calculate

### PR DESCRIPTION
**Pull request by Salvador-ON**
- [x] Create a new logic folder inside src and initialize two javascript files there: calculate.js and operate.js.
- [x] Implement the calculate.js module
- [x] Add the big.js package to your project dependencies.
- [x] Implement the calculate.js module
- [x] Implement the operate.js module
- [x] Operate.js should be used to calculate the total property.

### CODE REVIEW

**Comment by Salvador-ON** [comment]
I made a little more in this extra mile to test the calculations

live demo https://react-calculator-so.herokuapp.com/

**Comment by Salvador-ON** [comment]
Sticler is succesful
image

I was making some test with netlify to deploy and I'm having some troubles

**Review by fatymahmed — CHANGES_REQUESTED** [review]
## Hi @Salvador-ON,
You have done a good job in this milestone, however, there are a couple of things that need to changed for it to be approved.

**Check my comments below for a detailed review**

### Status: Changes Required ♻️
_Feel free to reach out to me on slack (Fatima) if you have any questions_

File: src/logic/operate.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+const Big = require('big.js');
+
+const operate = (ValueOne, ValueTwo, opt) => {
+  let result;
```
Comment (fatymahmed) [inline src/logic/operate.js:4]: - `valueOne` and `valueTwo` should be wrapped in Big class (you are only wrapping `valueOne`)

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    }
+  }
+
+  if (opt === '%') {
+    // if (!ValueTwo) {
+    result = Big(ValueOne).div(100);
+    // } else {
+    //   result = Big(ValueTwo).times(ValueOne).div(100);
+    // }
+  }
```
Comment (fatymahmed) [inline src/logic/operate.js:35]: - This operator should return the modulus of the two, like these `return valueOne.mod(valueTwo);
`

File: src/logic/operate.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+const operate = (ValueOne, ValueTwo, opt) => {
+  let result;
+  if (opt === '+') {
+    result = Big(ValueOne).plus(ValueTwo);
+  }
+
+  if (opt === '+/-') {
+    result = Big(ValueOne).plus(-1);
+  }
```
Comment (fatymahmed) [inline src/logic/operate.js]: This operator should be changing a positive number to a negative and vice versa by multiplying it to `-1`, you have done this in the `calculate.js` file, either remove it here or change the logic here then remove it in `calculate.js` file.

File: src/logic/calculate.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+  if (buttonName === '+/-') {
+    total = (currentValue * -1).toString();
+    currentValue = '';
+    operation = '';
+  }
+
+  if (buttonName === '%' && currentValue !== '') {
+    currentValue = (currentValue / 100).toString();
```
Comment (fatymahmed) [inline src/logic/calculate.js]: - You already have this in the operate function which you have imported, why not use that?

File: src/logic/calculate.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+const calculate = (dataObject, buttonName) => {
+  let { total, currentValue, operation } = dataObject;
+
+  total = total || '0';
+  const nums = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+  const ops = ['+', '-', 'x', '÷'];
+
+  if (buttonName === '.' && !currentValue.includes('.')) {
+    currentValue += buttonName;
+  }
```
Comment (fatymahmed) [inline src/logic/calculate.js:12]: should you update total if `.` is clicked on?

File: src/logic/calculate.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
+import operate from './operate';
+
+const calculate = (dataObject, buttonName) => {
+  let { total, currentValue, operation } = dataObject;
+
```
Comment (fatymahmed) [inline src/logic/calculate.js]: dataObject should contain `total`, `next`, `operation`, use the names provided on pathwright

File: src/logic/calculate.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  if (buttonName === '%' && currentValue !== '') {
+    currentValue = (currentValue / 100).toString();
+  }
+
+  if (ops.includes(buttonName)) {
+    if (
+      (buttonName === '-'
+        || buttonName === '+'
+        || buttonName === 'x'
+        || buttonName === '÷')
```
Comment (fatymahmed) [inline src/logic/calculate.js]: These 2 if statements seem to be performing the same check, if `buttonName` is included in `ops`, doesn't that mean that it is one of these operators?

File: src/logic/operate.js (hunk 1 above)
Comment (Salvador-ON) [reply src/logic/operate.js:4]: I wrapped value 2 in big class

File: src/logic/calculate.js (hunk 4 above)
Comment (Salvador-ON) [reply src/logic/calculate.js]: I made the proper modification is working from operate.js

File: src/logic/operate.js (hunk 3 above)
Comment (Salvador-ON) [reply src/logic/operate.js]: I fixed this issue moving to operate the logic

File: src/logic/calculate.js (hunk 7 above)
Comment (Salvador-ON) [reply src/logic/calculate.js]: i fix this issue

File: src/logic/calculate.js (hunk 6 above)
Comment (Salvador-ON) [reply src/logic/calculate.js]: i fix this iisue

**Review by Oghenebrume50 — APPROVED** [review]
### APPROVED 

Hi, @Salvador-ON kudos on making changes as outlined by the previous reviewer, please find merge and move on **This is approved** :+1: 

**RAPHAEL NORIODE** :mask:

**Comment by Salvador-ON** [comment]
thanks. I will proceed to merge

**Pull request merged**